### PR TITLE
Filtering lessons by class

### PR DESCRIPTION
**Pull request by loiswells97**

### CODE REVIEW

File: spec/features/lesson/listing_lessons_spec.rb [hunk 1]
```diff
@@ -69,6 +70,22 @@
     expect(data.size).to eq(1)
   end
 
+  it 'does not include lessons from outside the class if school_class_id provided' do
+    get("/api/lessons?school_class_id=#{school_class.id}", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
```
Comment (danhalson) [inline spec/features/lesson/listing_lessons_spec.rb:75]: I'd feel more confident in this step if there was a lesson associated with a school_class that isn't being returned, but it doesn't look like there is?

File: spec/features/lesson/listing_lessons_spec.rb [hunk 2]
```diff
@@ -69,6 +70,22 @@
...
+
+  it 'includes lessons from the class if school_class_id provided' do
+    lesson.update!(school_class_id: school_class.id)
+
+    get("/api/lessons?school_class_id=#{school_class.id}", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+
+    expect(data.size).to eq(1)
+  end
+
```
Comment (danhalson) [inline spec/features/lesson/listing_lessons_spec.rb:88]: also might be worth adding a step to check that there's no impact on the archived state of a lesson, so checking that if both `include_archived` and `school_class_id` are passed, an archived class is returned (and maybe the inverse?)

File: spec/features/lesson/listing_lessons_spec.rb (hunk 1 above)
Comment (loiswells97) [reply spec/features/lesson/listing_lessons_spec.rb:75]: Thanks Dan, added another test for this 👍

File: spec/features/lesson/listing_lessons_spec.rb (hunk 2 above)
Comment (loiswells97) [reply spec/features/lesson/listing_lessons_spec.rb:88]: Thanks Dan, have added a couple more test to reflect this ✅

**Review by sra405 — APPROVED** [review]
👍 nice changes + tests

**Pull request merged**